esm/utils/domains.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @typedef {import('./diag-domain-config.types.js').DomainInfo} DomainInfo
+ */
+
+/**
+ * Extracts `hostname`, `pathname`, `isWildcard` from a given `domain`
+ *
+ * @param {string} domain
+ * @return {{ hostname: string, pathname: string, isWildcard: boolean }}
+ */
+export function parseDomain (domain) {
+  const domainWithHttp = domain.match(/^https?:\/\//) != null ? domain : 'https://' + domain;
+  // With firefox, 'https://*.toto.com' is considered invalid so we strip it off for the test
+  // because we know this part is valid and we want the rest to be sanitized by the URL parser
+  const isWildcard = domain.startsWith('*.');
+
+  if (!isWildcard && domain.includes('*')) {
+    throw new DomainParseError('invalid-wildcard', 'Invalid wildcard format. "*" may only be used as a subdomain');
+  }
+
+  try {
+    const { hostname, pathname } = new URL(domainWithHttp.replace('*.', ''));
+
+    return {
+      hostname,
+      pathname,
+      isWildcard,
+    };
+  }
+  catch {
+    if (domain.length === 0) {
+      throw new DomainParseError('empty', 'Empty domain value');
+    }
+
+    throw new DomainParseError('invalid-format', 'Invalid domain format');
+  }
```

Comment:
Here you loose the cause of the error.

```suggestion
  catch (e) {
    if (domain.length === 0) {
      throw new DomainParseError('empty', 'Empty domain value');
    }

    throw new DomainParseError('invalid-format', 'Invalid domain format', e);
  }
```